src/fsharp/DependencyManager.Integration.fs

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+/// Helper members to integrate DependencyManagers into F# codebase
+module internal Microsoft.FSharp.Compiler.DependencyManagerIntegration
+
+open System
+open System.Reflection
+open System.IO
+open Microsoft.FSharp.Compiler.ErrorLogger
+
+// NOTE: this contains mostly members whose intents are :
+// * to keep ReferenceLoading.PaketHandler usable outside of F# (so it can be used in scriptcs & others)
+// * to minimize footprint of integration in fsi/CompileOps
+
+/// hardcoded to net461 as we don't have fsi on netcore
+let targetFramework = "net461"
```

Comment:
What is this actually used for? Can we not determine our current .Net Framework version instead of hardcoding?

Reply:
eventually yes. but not needed now

Reply:
Well what if I’m on .Net 4.5?

Reply:
then fsi is still...

Reply:
Right but this is going to be passed to nuget/paket/whatever to determine which assembly to reference... the precise .Net Framework version is important.

Reply:
yes you need to reference .net461 otherwise the fsi can't load it

Reply:
Hard coding net461 is wrong for ``FSharp.Compiler.Service.dll`` (which can be used to build tools that run on other frameworks) and for editing environments.  I'd rather see this passed down from the host tool, or otherwise configurable somehow.

Reply:
Of course hard coding is wrong, but it's a bit tricky since it needs to be the executing environment. Not necessarily the fcs environment. Consider we host the feature in fake

Reply:
Then ionide might run in net461 but fake process in netcore. Of course fake would restore a second time, but that would be a waste

Reply:
> Then ionide might run in net461 but fake process in netcore. Of course fake would restore a second time, but that would be a waste
 
I see.  

It is true that the "default assumed target framework" for the F# scripting model is net461. That is, an editing tool, or execution tool should process a script on the assumption it is targeting net461. 

For FSI on .NET Core this will be netcoreapp20 (we should at least define this conditionally here?), but editing tools will still have to assume net461 by default.

Because of this, we will likely need to add a directive ``#framework "net451"`` or ``#framework "netcoreapp20"`` I suppose, to allow editing tools to know that the script is really designed for use on .NET Core.

Reply:
Specification and switching of an alternate target framework is TBD.  The default target framework will either depend on the host or be netstandard2.0.